external/js/cothority/src/personhood/ro-pa-sci-instance.ts

```diff
@@ -121,15 +159,22 @@ export default class RoPaSciInstance extends Instance {
                 this.id,
                 RoPaSciInstance.contractID,
                 "second",
-                [
-                    new Argument({name: "account", value: coin.id}),
-                    new Argument({name: "choice", value: Buffer.from([choice % 3])}),
-                ],
+                args,
             ),
         );
         await ctx.updateCountersAndSign(this.rpc, [[signer], []]);
 
         await this.rpc.sendTransactionAndWait(ctx);
+        await this.update();
+        if (this.isCalypso()) {
+            const dreply = await lts.reencryptKey(await this.rpc.getProof(this.struct.calypsoWrite),
+                await this.rpc.getProof(this.struct.calypsoRead));
+            const preHash = await dreply.decrypt(priv);
+            this.firstMove = preHash[0];
+            this.fillUp = Buffer.alloc(31);
```

Comment:
If preHash size is known to be 32, you can allocUnsafe as each bit will be set anyway.

Reply:
I have no idea why I should use `allocUnsafe`. Just the name gives me the shivers...